specifyweb/frontend/js_src/lib/components/QueryBuilder/Wrapped.tsx

```diff
@@ -536,6 +545,8 @@ export function QueryBuilder({
                   getQueryFieldRecords={getQueryFieldRecords}
```

Comment:
Looking at the code now, it might be simpler and better to call `<QueryExportButtons>` inside of `<QueryResultsWrapper>` or even `<QueryResuts>`

the `resultsArray` thing is a cludge - you are storing the same state in two different places
(FYI: it doesn't really need to be a state, a ref would be fine - still, it should probably be avoided entirely)

can you try out and see if that makes things simpler? that would also add the csv export feature to express search (although, only when some search results are selected)